### PR TITLE
Contrast 36899 cli tool

### PR DESCRIPTION
**Pull request by kaitlynmortimer**
https://bitbucket.org/contrastsecurity/open-docs/pull-requests/388/added-new-page/diff

### CODE REVIEW

File: content/tools/cli/Cli-tool.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+-->
+  
+Use Contrast's command line interface (CLI) tool for software composition analysis (SCA) to gather and send library dependency information for your applications to the Contrast UI. 
+
+## How It Works
+
+The Contrast SCA CLI tool works in tandem with the Contrast UI to help you and your organization secure your application. For the Contrast SCA CLI tool to work with the UI, it must identify itself with a set of credentials, and also identify the application of which it's providing analysis.  
+
+The Contrast SCA is invoked by a command line interface. 
+
```
Comment (kaitlynmortimer) [inline content/tools/cli/Cli-tool.md]: Can we add a couple more sentences here about how the tool works?

File: content/tools/cli/Cli-tool.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+ --project-path applicationRootDirectory
+
+```
+
+<!-- 
+### Review 
+
+After you see a SUCCESS message you are now ready to view your dependency tree. (link to library hierarchy) 
+-->
+
```
Comment (kaitlynmortimer) [inline content/tools/cli/Cli-tool.md]: This is commented out until the library dependency docs are ready.

File: content/tools/cli/Cli-tool.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+ --api-key someAPIKey \ 
+
+ --authorization someAuthorizationCredentials \ 
+
+ --application-id someApplicationID \ 
+
+ --organization-id someOrganizationID \ 
+
+ --host ContrastUIDomain \
```
Comment (Giners) [inline content/tools/cli/Cli-tool.md]: The `--host` option isn't required per the current version. Maybe we ought to remove.

File: content/tools/cli/Cli-tool.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+The tool includes command line help, including the glossary of commands shown below, which you display with the `--h` option. 
+
+
+| Command                   | Description                                                                               |
+|---------------------------|-------------------------------------------------------------------------------------------|
+| --api-key string          | An agent API key provided by Contrast UI (**required**)                                 |
+| --authorization string    | Agent Authorization credentials provided by Contrast UI (**required**)                |
+| --organization-id string  | The ID of your organization in Contrast UI (**required**)                                 |
+| --application-id string   | The ID of the application cataloged by Contrast UI (**required**)                          |
+| --host string             | The name of the host and (optionally) the port expressed as <host>:<port> (**required**)|
```
Comment (Giners) [inline content/tools/cli/Cli-tool.md]: `--host` isn't required in the current version of the tool. It does default to the following value which we ought to document: `app.contrastsecurity.com`

File: content/tools/cli/Cli-tool.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+| --application-name string | The name of the application cataloged by Contrast UI (**optional**)                         |
```
Comment (Giners) [inline content/tools/cli/Cli-tool.md]: I think we ought to talk to Pauline and Annette and maybe exclude this option for now until we nail down our UX more about cataloguing applications from the CLI tool.

File: content/tools/cli/Cli-tool.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
+<!--
+title: "CLI Tool"
+description: "Instructions for installing the CLI tool for library analysis"
+tags: "tools cli library install"
+-->
+  
+Use Contrast's command line interface (CLI) tool for software composition analysis (SCA) to gather and send library dependency information for your applications to the Contrast UI.
```
Comment (Giners) [inline content/tools/cli/Cli-tool.md]: I think we ought to expand on this and improve the messaging around why someone would want to use the SCA tool. Our agents collect library dependency information for applications as well so we ought to distinguish the two.

What about something like the following...

Contrast's command line interface (CLI) tool can perform software composition analysis (SCA) on your application. The SCA CLI tool is able to gather data on your application while it is in its `development` phase that is typically not available when your application is in the `deployment` phase. Using the data captured by the SCA CLI tool we are able to provide you a more detailed view of your applications in the Contrast UI.



Thoughts?

File: content/tools/cli/Cli-tool.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+ --organization-id someOrganizationID \ 
+
+ --host ContrastUIDomain \ 
+
+ --project-path applicationRootDirectory
+
+```
+
+<!-- 
+### Review
```
Comment (Giners) [inline content/tools/cli/Cli-tool.md]: Is this something we need to add @kaitlynmortimer or you will add?

File: content/tools/cli/Cli-tool.md (hunk 2 above)
Comment (Giners) [reply content/tools/cli/Cli-tool.md]: I think they are ready. We wrote something to pass along to an internal customer that I was under the impression would evolve to this content. Is this what you understood as well?

File: content/tools/cli/Cli-tool.md (hunk 6 above)
Comment (Giners) [reply content/tools/cli/Cli-tool.md]: @kaitlynmortimer In regards to the following comment you left on line 14 of `content/tools/cli/Cli-tool.md`:

> Can we add a couple more sentences here about how the tool works?

Will my above comment/additional content work? Or are you thinking of something else?

File: content/tools/cli/Cli-tool.md (hunk 2 above)
Comment (kaitlynmortimer) [reply content/tools/cli/Cli-tool.md]: Yes, that's my understanding. I should've said until they're published on OD - then there will be a hyperlink to add here.

File: content/tools/cli/Cli-tool.md (hunk 7 above)
Comment (kaitlynmortimer) [reply content/tools/cli/Cli-tool.md]: It depends on the timing of publication.

File: content/tools/cli/Cli-tool.md (hunk 6 above)
Comment (kaitlynmortimer) [reply content/tools/cli/Cli-tool.md]: Yep, that looks good!

File: content/tools/cli/Cli-tool.md (hunk 1 above)
Comment (codebylogan) [reply content/tools/cli/Cli-tool.md]: How about adding
"The Contrast SCA CLI tool works in tandem with the Contrast UI to help you and your organization secure your application by giving you the visibility to understand how vulnerable libraries got introduced. When an open source library is added to an application, all of it’s dependencies are also inherited. Some of those indirect dependencies could be introducing vulnerabilities into your applications.  The Contrast SCA tool will identify all open source libraries and visualize these libraries as a hierarchical dependency tree.

For the Contrast SCA CLI tool to work with the UI, it must identify itself with a set of credentials, and also identify the application of which it's providing analysis.  "

File: content/tools/cli/Cli-tool.md (hunk 5 above)
Comment (codebylogan) [reply content/tools/cli/Cli-tool.md]: I agree we shouldnt point people to the fact they can catalogue an application this way.

I take it we just need to remove --application-name?

File: content/tools/cli/Cli-tool.md (hunk 6 above)
Comment (codebylogan) [reply content/tools/cli/Cli-tool.md]: Agree and good to point out its in the development stage

File: content/tools/cli/Cli-tool.md (hunk 5 above)
Comment (AnnetteNuge) [reply content/tools/cli/Cli-tool.md]: Fine with that.

File: content/tools/cli/Cli-tool.md (hunk 1 above)
Comment (kaitlynmortimer) [reply content/tools/cli/Cli-tool.md]: Incorporated some of this into the revised section here - some of it I saved for the library hierarchy article.

**Comment by gdvdsn-contrast** [comment]
This doc has been updated to include Damian's changes. We can use this draft for final editing and approval. Please feel free to make changes or suggestions! @AnnetteNuge

**Comment by gdvdsn-contrast** [comment]
Contrast CLI has been published here: https://docs.contrastsecurity.com/tools-cli.html#contrast-cli

**Pull request merged**